contrib/fmhash/fmhash.c

```diff
@@ -197,10 +195,9 @@ hash_wrapper3(struct svar *__restrict__ const sourceVal, struct svar *__restrict
 		FINALIZE;
 	}
 
-	CHKiRet((hcontext->hash_wrapper_1_2(sourceVal, seedVal, hcontext)));
-	xhash = hcontext->xhash % mod;
-	DBGPRINTF("fmhash: hashXXmod generated hash-mod %" PRIu64 ".", xhash);
-	hcontext->xhash = xhash;
+	CHKiRet((hcontext->hash_wrapper_1_2(sourceVal, seedVal, hcontext, xhash)));
+	(*xhash) = (*xhash) % mod;
```

Comment:
cppcheck (via Codacy) has a false positive here, it does not seem to understand that the check in 191 protects again `mod==0`. Nevertheless, we should try to work around this issue. How about using `mod = 1` after line 193?

Reply:
Adding mod =1 made clang analyzer angry. Adding if(mod!=0) did the trick.

Reply:
> Adding mod =1 made clang analyzer angry. Adding if(mod!=0) did the trick.

oh, the joys of static analyzers... ;-) Thanks for taking care of this. Will merge most probably tomorrow (now  I really need to run out of office ;-)